Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/RenderPass.h

```diff
@@ -153,7 +153,7 @@ namespace AZ
 
             // List of all ShaderResourceGroups to be bound during rendering or computing
             // Derived classed may call BindSrg function to add other srgs the list
-            ShaderResourceGroupList m_shaderResourceGroupsToBind;
+            AZStd::unordered_map<uint8_t, const RHI::ShaderResourceGroup*> m_shaderResourceGroupsToBind;
```

Comment:
This changes from no memory allocations to num entries + 1, How bad of an error does it fix?

Reply:
would fixed_unordered_map work?

Reply:
Basically the BindSrg became a public function due a special need from terrain FP. 
We can't control how often the function is called so it might be easily broken if it's called many time before the pass is rendered. 
it's not the best fix since I don't like that it won't be cleared when a pass is not rendered because any reasons. But it's a ok solution for the current problem. 
I am collecting similar problems for the pass system so I may come back for a better solution for these kind of problems.